tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -378,6 +390,18 @@ func (iptConfigurator *IptablesConfigurator) run() {
 	// 127.0.0.6 is bind connect from inbound passthrough cluster
 	iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-o", "lo", "-s", "127.0.0.6/32", "-j", constants.RETURN)
 
+	// for inbound traffic envoy redirect to specific local pod IP, just RETURN.
+	// these rules are avoid for redirecting traffic into ISTIO_IN_REDIRECT for inbound traffic
+	for _, uid := range split(iptConfigurator.cfg.ProxyUID) {
+		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT,
+			"-s", "127.0.0.1/32", "!", "-d", "127.0.0.1/32", "-m", "owner", "--uid-owner", uid, "-j", "RETURN")
+	}
```

Comment:
This conflicts with L409 below

Reply:
L409 handle traffic `app calls back to itself via Envoy when using the service VIP`
the envoy will send request like `localPodIP:randomPort -> localPodIP:port`, L409 catch this traffic and redirect to envoy.

when using `bindPodIPPorts` feature, envoy will send inbound traffic like `127.0.0.1:randomPort -> localPodIP:port`. without the new added rules, this traffic will be handled by L409 and cause an infinite loop.

Reply:
How could the src ip is `127.0.0.1`?

Reply:
Is this guaranteed from envoy implementation?

Reply:
> How could the src ip is `127.0.0.1`?

changes in cluster.go `localCluster.UpstreamBindConfig` is used to make this possible